controllers/volsync/vshandler.go

```diff
@@ -1571,3 +1662,376 @@ func ValidateObjectExists(ctx context.Context, c client.Client, obj client.Objec
+		addLabel(lrd, VolSyncDoNotDeleteLabel, VolSyncDoNotDeleteLabelVal)
+
+		pvcAccessModes := []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce} // Default value
+		if len(rdSpec.ProtectedPVC.AccessModes) > 0 {
+			pvcAccessModes = rdSpec.ProtectedPVC.AccessModes
+		}
+
+		lrd.Spec.RsyncTLS = &volsyncv1alpha1.ReplicationDestinationRsyncTLSSpec{
+			ServiceType: v.getRsyncServiceType(),
+			KeySecret:   &pskSecretName,
+
+			ReplicationDestinationVolumeOptions: volsyncv1alpha1.ReplicationDestinationVolumeOptions{
+				CopyMethod:       volsyncv1alpha1.CopyMethodDirect,
+				Capacity:         rdSpec.ProtectedPVC.Resources.Requests.Storage(),
+				StorageClassName: rdSpec.ProtectedPVC.StorageClassName,
+				AccessModes:      pvcAccessModes,
+				DestinationPVC:   &rdSpec.ProtectedPVC.Name,
+			},
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	// Now check status - only return an RD if we have an address filled out in the ReplicationDestination Status
+	if lrd.Status == nil || lrd.Status.RsyncTLS == nil || lrd.Status.RsyncTLS.Address == nil {
+		l.V(1).Info("Local ReplicationDestination waiting for Address ...")
+
+		return nil, nil
+	}
+
+	l.V(1).Info("Local ReplicationDestination Reconcile Complete", "op", op)
+
+	return lrd, nil
+}
+
+//nolint:funlen
+func (v *VSHandler) reconcileLocalRS(rd *volsyncv1alpha1.ReplicationDestination,
+	rsSpec *ramendrv1alpha1.VolSyncReplicationSourceSpec,
+	pskSecretName, address string,
+) (*volsyncv1alpha1.ReplicationSource, error,
+) {
+	storageClass, err := v.getStorageClass(rsSpec.ProtectedPVC.StorageClassName)
+	if err != nil {
+		return nil, err
+	}
+
+	volumeSnapshotClassName, err := v.getVolumeSnapshotClassFromPVCStorageClass(storageClass)
+	if err != nil {
+		return nil, err
+	}
+
+	// Fix for CephFS (replication source only) - may need different storageclass and access modes
+	err = v.ModifyRSSpecForCephFS(rsSpec, storageClass)
+	if err != nil {
+		return nil, err
+	}
+
+	pvc, err := v.setupLocalRS(rd)
+	if err != nil {
+		return nil, err
+	}
+
+	lrs := &volsyncv1alpha1.ReplicationSource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getLocalReplicationName(rsSpec.ProtectedPVC.Name),
+			Namespace: v.owner.GetNamespace(),
+		},
+	}
+
+	op, err := ctrlutil.CreateOrUpdate(v.ctx, v.client, lrs, func() error {
+		if err := ctrl.SetControllerReference(v.owner, lrs, v.client.Scheme()); err != nil {
+			v.log.Error(err, "unable to set controller reference")
+
+			return err
+		}
+
+		addVRGOwnerLabel(v.owner, lrs)
+
+		lrs.Spec.Trigger = &volsyncv1alpha1.ReplicationSourceTriggerSpec{
+			Manual: pvc.GetName(),
+		}
+
+		lrs.Spec.SourcePVC = pvc.GetName()
+		lrs.Spec.RsyncTLS = &volsyncv1alpha1.ReplicationSourceRsyncTLSSpec{
+			KeySecret: &pskSecretName,
+			Address:   &address,
+
+			ReplicationSourceVolumeOptions: volsyncv1alpha1.ReplicationSourceVolumeOptions{
+				CopyMethod:              volsyncv1alpha1.CopyMethodDirect,
+				VolumeSnapshotClassName: &volumeSnapshotClassName,
```

Comment:
I think VolumeSnapshotClassName, StorageClassName, AccessModes may not be required since we're using direct mode here - just leads to confusion as I'm unsure which volumesnapshotclass we're even passing in (rsSpec.ProtectedPVC is which pvc?)

Reply:
Ok, I can remove those.